src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.netcoreapp.cs

```diff
@@ -0,0 +1,498 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Pkcs.Tests
+{
+    public static partial class SignedCmsTests
+    {
+        [Fact]
+        public static void SignCmsUsingExplicitRSAKey()
```

Comment:
It feels like these tests are complicatedly set up.

I'd expect the whole body of this to be something like

```C#
[Fact]
public static void SignCmsUsingExplicitRSAKey()
{
    using (X509Certificate2 cert = Certificates.RSA2048SignatureOnly.GetCertificate())
    using (RSA privKey = cert.GetRSAPrivateKey())
    {
        VerifyWithExplicitPrivateKey(cert, privKey);
    }
}

private static void VerifyWithExplicitPrivateKey(X509Certificate2 cert, AsymmetricAlgorithm key)
{
    using (var pubCert = new X509Certificate2(cert.RawData))
    {
        byte[] content = { 9, 8, 7, 6, 5 };
        ContentInfo contentInfo = new ContentInfo(content);
        SignedCms cms = new SignedCms(contentInfo);
  
        CmsSigner signer = new CmsSigner(someIdType, pubCert, key)
        {
             IncludeOption = X509IncludeOption.EndCertOnly,
             DigestAlgorithm = new Oid(Oids.Sha1, Oids.Sha1).
        };

        cms.ComputeSignature(signer);

        Assert.Equal(1, cms.SignerInfos.Count);
        Assert.True(cms.CheckSignature(true));
    }
}
```

And a similar utility method for counter-signing (which could start with an existing test document).